sklearn/pipeline.py

```diff
@@ -869,6 +869,26 @@ def __init__(
         self.verbose = verbose
         self._validate_transformers()
 
+    def __getitem__(self, ind):
+        """Returns a single estimator in the feature union.
+
+        Indexing with an integer or string will return an estimator.
+        """
+        if isinstance(ind, slice):
+            raise ValueError("FeatureUnion slicing not supported")
+
+        try:
+            name, est = self.transformer_list[ind]
+        except TypeError:
+            # Not an int, try get transformer by name
+            return self.named_transformer_list[ind]
+        return est
```

Comment:
I don't think we need `__getitem__` for feature union. Index-based access in the `Pipeline` case is natural because the order of the transformer is meaningful for the `Pipeline` but it not that meaningful for `FeatureUnion`.

Named based lookup using the named transformers property is enough (and more explicit) for model inspection purposes.